### ISSUE TITLE
[Docs] the "random" module should not be imported by default

### ISSUE THREAD

**Issue by forFudan** [modularml/mojo#3076]
### Where is the problem?

https://docs.modular.com/mojo/stdlib/random/random/

### What can we do better?

According to the official documentation, the `random` module needs to be imported before use.

> You can import these APIs from the random package. For example:
> 
> ```
> from random import seed
> ```

But the following code work without error:

```
fn main():
    var arr = random.randn_float64()
    print(arr)
```

I would prefer to make the module to be not imported by default, being consistent with the documentation, and makes the naming space cleaner.

**Comment by gabrieldemarmiesse** [comment]
Thanks for the issue. I think it's a duplicate of https://github.com/modularml/mojo/issues/2529 . Can you confirm? If yes, I believe this issue can be closed and we'll keep https://github.com/modularml/mojo/issues/2529 opened as it's a more general issue of the relationship between the compiler and the stdlib.

**Comment by forFudan** [comment]
> Thanks for the issue! I think it's a duplicate of #2529 . Can you confirm? If yes, I believe this issue can be closed and we'll keep #2529 opened as it's a more general issue of the relationship between the compiler and the stdlib.

I think they are different issues. The example in #2529 is conditional on the fact that someone, by mistake, imports `now()` into any of the builtin files. However, the `random` module is always imported automatically, even without user changing the builtin files.

**Comment by JoeLoser** [comment]
This is indeed a duplicate of https://github.com/modularml/mojo/issues/2529.  Specifically, in this case, all of the builtin files under the `builtin` module, **and their transitive closure set of imports** are automatically imported into every Mojo program.  In this case, https://github.com/modularml/mojo/blob/bc18cb454cd1bf7384da6eb86f79907b589c2419/stdlib/src/builtin/hash.mojo#L28 imports `random`, making `random` module available to every Mojo program.  This is not great.
